### PR TITLE
Switch to jupyter_server

### PR DESCRIPTION
**Pull request by SylvainCorlay**
Requires initial release of `jupyter_server`.

### CODE REVIEW

**Comment by blink1073** [comment]
Whoa, is jupyter_server coming soon??

**Comment by SylvainCorlay** [comment]
I am getting back into it since yesterday.

I got JupyterLab to run on the top of it with this PR and the one on jupyterlab...

**Comment by SylvainCorlay** [comment]
OK, let me know what you think. This falls back on `jupyter_server` when notebook is not available. 

This makes `pip uninstall notebook` a way to test `jupyter_server`.

**Comment by SylvainCorlay** [comment]
Comes together with https://github.com/jupyterlab/jupyterlab/pull/4954

**Comment by blink1073** [comment]
Confirmed that this works, but we'll need to switch `setup.py` to use `jupyter_server` in both packages (I had to do so manually to get `pip` to stop complaining).

**Comment by blink1073** [comment]
Tested with nothing installed, then just `notebook`, then just `jupyter_server`, then with both.

**Comment by SylvainCorlay** [comment]
Yep, I don't know if there is a notion of fallback package with pypi, or something like this...

**Comment by SylvainCorlay** [comment]
@blink1073 another proposal would be drop the `notebook` dependency in `jupyterlab`, and import everything in a Shim defined in `jupyterlab_launcher`...

Depending on a runtime option in `jupyterlab_launcher`, the shim could be one or the other?

**Comment by SylvainCorlay** [comment]
In any case, `jupyterlab` could already drop the `notebook` requirement which is already provided as a dependency of `jupyterlab_launcher`.

**Comment by SylvainCorlay** [comment]
So here I am merely importing what is used by jupyterlab into `server`, instead of using tools like ShimModule from IPython which seems a bit overkill.

**Comment by SylvainCorlay** [comment]
@blink1073 let me know if this is fine with you!

File: jupyterlab_launcher/servertest.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Shim for notebook server or jupyter_server
+#
+# Provides:
+#  - ServerTestBase
+
```
[the remark this answers is not in the document]
Comment (SylvainCorlay) [reply jupyterlab_launcher/servertest.py]: done

**Comment by afshin** [comment]
@SylvainCorlay FYI There was a merge conflict so I rebased.

**Comment by afshin** [comment]
Also, what do you think about the name of this package ... it seems like it is more natural to call it `jupyterlab_server` in light of where we are headed.

**Comment by blink1073** [comment]
Launcher is a confusing term for what this library does, :+1: for "server".

**Comment by blink1073** [comment]
I'm going to merge this so I can start the Python 3 migration without causing a bunch of conflicts.  Once `jupyter_server` is released we can update the dependencies.

**Pull request merged**